toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -214,10 +214,17 @@ func fstabEntriesToMountPoints(fstabEntries []diskutils.FstabEntry, diskPartitio
 			return nil, nil, nil, err
 		}
 
-		// Unset read-only flag so that read-only partitions can be customized.
-		// Unset noexec flag so that if rootfs is set as noexec, image can still be customized. For example, allowing
-		// grub2-mkconfig to be called.
-		vfsOptions := fstabEntry.VfsOptions & ^diskutils.MountFlags(unix.MS_RDONLY|unix.MS_NOEXEC)
+		var vfsOptions diskutils.MountFlags
+		if overrideReadonly {
+			// Unset read-only flag so that read-only partitions can be customized.
+			// Unset noexec flag so that if rootfs is set as noexec, image can still be customized. For example, allowing
+			// grub2-mkconfig to be called.
+			vfsOptions = fstabEntry.VfsOptions & ^diskutils.MountFlags(unix.MS_RDONLY|unix.MS_NOEXEC)
+		} else {
+			// In post-customization scenarios (e.g. when the image is re-mounted for injection),
+			// we may want to respect the original fstab flags, so the override is disabled.
+			vfsOptions = fstabEntry.VfsOptions
```

Comment:
You should set all the partitions to be read-only, since you aren't planning on changing anything on the disk. You are just reading info.

Also, I would rename `overrideReadonly` to just `readonly` (which will reverse the meaning).